test/api.test.ts

```diff
@@ -73,4 +73,46 @@ suite(`AzureFunctionsExtensionApi`, () => {
         );
         await validateProject(folderPath, validateOptions);
     });
+
+    test('createFunction dotnet with targetFramework', async () => {
+        const functionName: string = 'endpoint1';
+        const language: string = ProjectLanguage.CSharp;
+        const workspaceFolder = getTestWorkspaceFolder();
+        const projectSubpath = 'api';
+        const folderPath: string = path.join(workspaceFolder, projectSubpath);
+
+        await runWithInputs('api.createFunction', [language, /6/i, 'Company.Function', 'Anonymous'], registerOnActionStartHandler, async () => {
+            await api.createFunction({
+                folderPath,
+                functionName,
+                templateId: 'HttpTrigger',
+                languageFilter: /^C\#$/i,
+                functionSettings: { authLevel: 'anonymous' },
+                targetFramework: ['net8.0', 'net7.0', 'net6.0']
+            });
+        });
+
+        const validateOptions: IValidateProjectOptions = getCSharpValidateOptions('net6.0', FuncVersion.v4);
+        await validateProject(folderPath, validateOptions);
+    });
+
+    // When passing in a language version that is not in the target framework the last two inputs should not be prompted if they are this test will fail
```

Comment:
This test is ensuring that target framework properly filters the picks from all available framework versions.

From this comment, it's not clear to me:
1. Why `api.createFunction` doesn't throw here, since selecting version 8 shouldn't be possible
2. What last two inputs do you mean, do you mean the next steps in the createFunction wizard?
3. How do you know this test will fail if 8 is a valid pick?

I think a better comment would start like:

// Intentionally pass a version (8) that hasn't been specified in `targetFramework` (6 & 7) to verify it isn't a possible pick. In the correct case (when 8 isn't a pick)... And in the incorrect case (when 8 is available)...